sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1432,6 +1471,13 @@ def task_worker(self):
                         self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_READY
                         continue
 
+                    host_lanes = self.get_cmis_host_lane_mask(api, host_lane_count, channel, speed)
+                    if host_lanes == 0:
+                        self.log_error("{}: Invalid lane mask received host_lane_count {} channel {} speed {}!".format(
+                                        lport, host_lane_count, channel, speed))
+                        self.port_dict[lport]['cmis_state'] = self.CMIS_STATE_FAILED
```

Comment:
CMIS_STATE_FAILED, I dont see this state is handled, so what is the expectation when SW set this state.
Do we need to bring the link down/move to DP-Deactivated if the state is already DP_Activation /DP_initialized?

Reply:
@jaganbal-a - Once a port is in CMIS_STATE_FAILED, we don't want to pass any config to the module and hence, any future config will be ignored when a port is in this state.
I am marking the state to failed before starting the CMIS initialization so we need not take care of other CMIS init relevant steps. Also, in this case, since we don't have the host_lane count, it is not possible to take any further action.

Reply:
I have filed #344 to track this.